### ISSUE TITLE
ejabberd plugin does not work with ejabberd 14

### ISSUE THREAD

**Issue by PF4Public** [munin-monitoring/munin#488]
In a nutshell problem is "This command can only be run by root or the user ejabberd".
This started after I've upgraded to ejabberd 14.
It worked well before however.
It is not a munin bug really, but I could not find better place to post.
What I've already done to identify the source:
1. "user root" and "user ejabberd" in plugin config
2. As expected putting `$(whoami)` at the beginning of "ejabberd_" being run by `munin-run ejabberd_users` shows either root or ejabberd respectively.
3. Tried adding following configs, but no success either:

> env.HOME /var/lib/ejabberd
> env.USER ejabberd
> env.USERNAME ejabberd
> env.MAIL ejabberd
> env.LOGNAME ejabberd
> env.MUNIN_PLUGINUSER ejabberd
> env.MUNIN_USER ejabberd

`sudo -u ejabberd ejabberdctl` runs fine from cli, but not from munin plugin. 
Don't know what else I can do to isolate the reason for this weird (or maybe not) behaviour.

**Comment by kero242** [comment]
still nox fix? :-(

**Comment by PF4Public** [comment]
The following code
```
echo $(whoami)
echo
echo $(id -G)
echo
echo $(id -g)
echo
echo $(id -g ejabberd)
echo
echo $(id -g munin)
```
produces this output:

> ejabberd
> 
> 139
> 
> 139
> 
> 123
> 
> 139

So, that even though munin has changed user to ejabberd, considering the id output, it didn't.

**Comment by shapirus** [comment]
Have you tried setting the HOME environment variable to the directory containing the .erlang.cookie file? I've found that to be important while for example sudo-ing to ejabberd user to run ejabberdctl. It's quite likely that munin-node doesn't set HOME to the expected value.

**Comment by shapirus** [comment]
I can also suggest comparing the output of "env" when run from under munin-node and sudo -u ejabberd.

**Comment by PF4Public** [comment]
>Have you tried setting the HOME environment variable to the directory containing the .erlang.cookie file?

See my first post:
>Tried adding following configs, but no success either:
>env.HOME /var/lib/ejabberd

Still I believe that main cause is not setting gid by munin. At least it could be the first obstacle.
Should I fool `ejabberdctl` by replacing `-d` with `-u` in `id` checks it no longer complains and seems to pass given arguments forward to `erl`, however this still fails with a bunch of acceess denied like: 

> sed: can't read /etc/ejabberd/ejabberd.yml: Permission denied

permissions are as follows:

>-rw-------   1 ejabberd ejabberd    3404 May 19  2015 ejabberd.yml

this:

>-rw-r--r--   1 ejabberd ejabberd    3404 May 19  2015 ejabberd.yml

does not help either.

Therefore I think that gid check (`-g`) in `ejabberdctl` was made on purpose.
Would it be possible for munin to change gid as well as uid?

**Comment by PF4Public** [comment]
I've filled [ejabberd bugreport](https://github.com/processone/ejabberd/issues/958), maybe they could help us resolving this.

**Comment by PF4Public** [comment]
For anyone interested lengthy explanation follows.
**TL;DR:** 
Munin does not change group properly, only appends it to GID list or so it seems (is it a candidate for another bugreport?), `ejabberdctl` however does check for proper GID. Unless munin changes its group handling we have 2 options:
* use
>user root
 group root

* append `EXEC_CMD="bash -c"` after `EXEC_CMD="false"` in `ejabberdctl` and use
>user ejabberd
 group ejabberd

Be warned though that the latter may have implications 
**TL;DR: end** 

Here is the missing part of this puzzle:
Setting 
>user root
 group root

Effectively adds GID=0 to GID list, which via
`if [ $GID -eq 0 ] ; then`
mandates 
`EXEC_CMD="su $INSTALLUSER -c"`
making it all work fine. Crucial thing here is that both user and group defined. Group bypasses check for effective GID to be equal to ejabberd, while user root allows for succesfull `su`

Using ejabberd is different in that munin does not change effective GID, instead it just adds it to GID list or so it seems, thus GID check fails. However for this particular use having ejabberd GID in GID list is sufficient to get statistics out of ejabberd, so adding `EXEC_CMD="bash -c"` right after `EXEC_CMD="false"` in `ejabberdctl` does the trick.

**Comment by sumpfralle** [comment]
Thankfully the ejabberd people fixed the [issue](https://github.com/processone/ejabberd/issues/958).
